hail/src/main/scala/is/hail/expr/ir/NormalizeNames.scala

```diff
@@ -1,9 +1,49 @@
 package is.hail.expr.ir
 
 import is.hail.backend.ExecuteContext
+import is.hail.expr.ir.NormalizeNames.needsRenaming
+import is.hail.types.virtual.Type
 import is.hail.utils.StackSafe._
 
-import scala.annotation.nowarn
+import scala.collection.mutable
+
+object NormalizeNames {
+  def apply[T <: BaseIR](
+    ctx: ExecuteContext,
+    ir: T,
+    env: BindingEnv[String] = BindingEnv.empty.createAgg.createScan,
+    normFunction: Int => String = iruid(_),
+    allowFreeVariables: Boolean = false,
+  ): T = {
+    val noSharing = ir.noSharing(ctx)
+    val _ = new NormalizeNames(normFunction, allowFreeVariables).normalizeIROld(
+      noSharing,
+      env,
+    ).run().asInstanceOf[T]
+    val old = new NormalizeNames(iruid(_), allowFreeVariables).normalizeIROld(
+      noSharing,
+      env,
+    ).run().asInstanceOf[T]
+    val new_ = new NormalizeNames(iruid(_), allowFreeVariables).normalizeIROld(
+      noSharing,
+      env,
+    ).run().asInstanceOf[T]
+//    if (env.allEmpty) TypeCheck(ctx, res.asInstanceOf[BaseIR])
+    assert(new_ == old, s"new:\n${Pretty.sexprStyle(new_)}\nold:\n${Pretty.sexprStyle(old)}")
```

Comment:
wot - why are you running this three times?

Reply:
This is asserting that the old and new implementations agree. Thanks for the reminder to make a PR removing the old implementation.

The third pass is because I was running into issues with the `normFunction` being used. But this is currently returning a result that ignores the requested norm function and just uses the default. If that passes ci, I'll probably just remove that argument. I can do that in this PR.